build.gradle

```
@@ -466,6 +466,10 @@ subprojects {
     if (shouldUseJUnit5) {
       useJUnitPlatform {
         includeTags "integration"
+        // KAFKA-14109
+        // Both engines are needed to run JUnit 4 tests alongside JUnit 5 tests.
+        // junit-vintage (JUnit 4) can be removed once the JUnit 4 migration is complete.
+        includeEngines "junit-vintage", "junit-jupiter"
       }
```

I experimented a bit and found the following solution.
```suggestion
      if (project.name.equals('streams')) {
        useJUnitPlatform {
          includeTags "integration"
          includeTags 'org.apache.kafka.test.IntegrationTest'
          includeEngines "junit-vintage", "junit-jupiter"
        }
      } else {
        useJUnitPlatform {
          includeTags "integration"
        }
      }
```
In this way, we can limit the change to Streams.
The caveat is that we need to replace `@Category({IntegrationTest.class})` with `@Tag{"integration"}` in the integration tests that have already been migrated to JUnit 5.

---

Made the changes as suggested. Please see latest revision.